src/main/sarl/fr/utbm/info/vi51/worldswar/simulator/Simulator.sarl

```diff
@@ -24,36 +27,52 @@ agent Simulator {
 	private final val STARTUP_DELAY = 2000
 	
 	private var controller : Controller
-	private var gui : FakeGUI
+	private var gui : Window
 	
 	private var listeners : List<SimulatorListener>
-	
+
 	/** Simulation speed, as chosen by the user */
 	private var simulationSpeed = SimulationSpeed.NORMAL
 	
 	/** Indicates whether the step timer has passed, hence a new step should be fired */
 	private var stepToFire = false
 	/** Indicates whether the environment is ready to compute a new step */
 	private var environmentReady = false
+	/** Indicates if the simulation is currently running */
+	private var simulationRunning = false
+	/** Indicates if the simulation is currently stopping */
 	
+	private var newSimulationPending = false
 	
 	on Initialize {
 		controller = new Controller(defaultSpace)
-		gui = new FakeGUI(controller)
-		spawn(typeof(Environment))
+		gui = new Window(controller)
 		
 		listeners = new LinkedList
 		listeners.add(gui)
-
-		in(STARTUP_DELAY) [ startSimulation ]
+		
 	}
 	
-	private def startSimulation {
-		environmentReady = true
-		stepToFire = true
-		fireSimulationStep
+	private def newSimulation {
+		if(simulationRunning) {
+			stopSimulation
+			newSimulationPending = true
+		} else {
+			simulationRunning = true
+			newSimulationPending = false
+			spawn(typeof(Environment))
+			environmentReady = true
+			stepToFire = true
+			fireSimulationStep
```

Comment:
STARTUP_DELAY n'est plus utilisée, il faudrait soit la supprimer soit mettre `in(STARTUP_DELAY)[fireSimulationStep]` ici